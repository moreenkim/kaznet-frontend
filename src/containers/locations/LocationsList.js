import React, { Component } from 'react';
import { connect } from 'react-redux';

import * as locationActions from '../../store/locations/actions';
import * as locationSelectors from '../../store/locations/reducer';
import * as globalActions from '../../store/global/actions'

import ListView from '../../components/ListView';
import ElementMap from '../ElementMap';


class LocationsList extends Component {

    componentDidMount() {
        this.props.dispatch(locationActions.fetchLocations());
        this.props.dispatch(globalActions.changePageTitle('Locations'));
        this.props.dispatch(globalActions.changePageTitleButton('+ Create Location'))
    }

    render() {
        if (!this.props.rowsById) return this.renderLoading();
        return (
            <div className="LocationList">
                <ListView
                    renderHeaders={this.renderHeaders}
                    rowsIdArray={this.props.rowsIdArray}
                    rowsById={this.props.rowsById}
                    renderRow={this.renderRow} />
            </div>
        );
    }

    renderLoading() {
        return (
            <p>Loading...</p>
        );
    }

    renderRow(row) {
        const rowItems = [
            row.attributes.name,
            row.attributes.parent_name,
            row.attributes.location_type_name,
            row.attributes.description
        ];
        return (
            <ElementMap items={rowItems} HTMLTag='td' />
        );
    }

    renderHeaders() {
        const headerItems = [
          'Name',
          'Parent',
          'Type',
          'Description',
        ];
        return (
          <ElementMap items={headerItems} HTMLTag='th' />
        );
      }
}

function mapStateToProps(state) {
    return {
        rowsById: locationSelectors.getLocationsById(state),
        rowsIdArray: locationSelectors.getLocationsIdArray(state)
    };
}

export default connect(mapStateToProps)(LocationsList);