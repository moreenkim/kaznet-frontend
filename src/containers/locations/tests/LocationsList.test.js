import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import { Router } from "react-router";
import createBrowserHistory from "history/createBrowserHistory";

import { LocationsList } from "../LocationsList";
import * as fixtures from "../../../store/locations/tests/fixtures";

const history = createBrowserHistory();

describe("containers/locations/LocationList", () => {
  it("renders without crashing", () => {
    shallow(
      <LocationsList
        fetchLocations={function() {}}
        changePageTitle={function() {}}
        changePageTarget={function() {}}
        changePageTitleButton={function() {}}
        changePageNumber={function() {}}
        searchVal={function() {}}
        showListTitle={function() {}}
        location={history.location}
        searchParam={""}
      />
    );
  });

  it("renders location list correctly", () => {
    const wrapper = mount(
      <Router history={history}>
        <LocationsList
          fetchLocations={function() {}}
          changePageTitle={function() {}}
          changePageTarget={function() {}}
          changePageTitleButton={function() {}}
          changePageNumber={function() {}}
          searchVal={function() {}}
          showListTitle={function() {}}
          rowsById={fixtures.locationById}
          rowsIdArray={fixtures.locationIdArray}
          endpoint={"locations"}
          pageLinks={fixtures.pageLinks}
          totalPages={fixtures.totalPages}
          currentPage={fixtures.currentPage}
          firstPage={fixtures.firstPage}
          lastPage={fixtures.lastPage}
          location={history.location}
          searchParam={""}
        />
      </Router>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
