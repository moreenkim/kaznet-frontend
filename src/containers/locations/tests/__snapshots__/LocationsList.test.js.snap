// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/locations/LocationList renders location list correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <LocationsList
    changePageTarget={[Function]}
    changePageTitle={[Function]}
    changePageTitleButton={[Function]}
    currentPage={1}
    endpoint="locations"
    fetchLocations={[Function]}
    firstPage={1}
    lastPage={2}
    pageLinks={
      Object {
        "first": "http://localhost:8000/api/v1/locations/?page=1",
        "last": "http://localhost:8000/api/v1/locations/?page=1",
        "next": "http://localhost:8000/api/v1/locations/?page=2",
        "prev": null,
      }
    }
    rowsById={
      Object {
        "attributes": Object {
          "country": "CK",
          "created": "2018-07-09T14:53:09.732230+03:00",
          "description": "Something",
          "geopoint": null,
          "location_type_name": null,
          "modified": "2018-07-09T14:53:09.732261+03:00",
          "name": "Sol Point",
          "parent_name": null,
          "radius": null,
          "shapefile": null,
        },
        "id": "7",
        "relationships": Object {
          "location_type": Object {
            "data": null,
          },
          "parent": Object {
            "data": null,
          },
        },
        "type": "Location",
      }
    }
    showListTitle={[Function]}
    totalPages={2}
  >
    <div
      className="LocationList"
    >
      <ListView
        currentPage={1}
        endpoint="locations"
        firstPage={1}
        lastPage={2}
        pageLinks={
          Object {
            "first": "http://localhost:8000/api/v1/locations/?page=1",
            "last": "http://localhost:8000/api/v1/locations/?page=1",
            "next": "http://localhost:8000/api/v1/locations/?page=2",
            "prev": null,
          }
        }
        renderHeaders={[Function]}
        renderRow={[Function]}
        rowsById={
          Object {
            "attributes": Object {
              "country": "CK",
              "created": "2018-07-09T14:53:09.732230+03:00",
              "description": "Something",
              "geopoint": null,
              "location_type_name": null,
              "modified": "2018-07-09T14:53:09.732261+03:00",
              "name": "Sol Point",
              "parent_name": null,
              "radius": null,
              "shapefile": null,
            },
            "id": "7",
            "relationships": Object {
              "location_type": Object {
                "data": null,
              },
              "parent": Object {
                "data": null,
              },
            },
            "type": "Location",
          }
        }
        totalPages={2}
      >
        <div>
          <Table
            bordered={true}
            className="kaznet-table"
            responsiveTag="div"
            tag="table"
          >
            <table
              className="kaznet-table table table-bordered"
            >
              <thead>
                <tr>
                  <ElementMap
                    HTMLTag="th"
                    items={
                      Array [
                        "Name",
                        "Parent",
                        "Type",
                        "Description",
                      ]
                    }
                  >
                    <th
                      key="0"
                    >
                      Name
                    </th>
                    <th
                      key="1"
                    >
                      Parent
                    </th>
                    <th
                      key="2"
                    >
                      Type
                    </th>
                    <th
                      key="3"
                    >
                      Description
                    </th>
                  </ElementMap>
                </tr>
              </thead>
              <tbody />
            </table>
          </Table>
          <Pagination
            aria-label="Page navigation example"
            listTag="ul"
            tag="nav"
          >
            <nav
              aria-label="Page navigation example"
              className=""
            >
              <ul
                className="pagination"
              >
                <PaginationItem
                  disabled={false}
                  tag="li"
                >
                  <li
                    className="page-item"
                  >
                    <Link
                      aria-label="First"
                      className="page-link"
                      replace={false}
                      to="/locations/?search=undefined&page=1"
                    >
                      <a
                        aria-label="First"
                        className="page-link"
                        href="/locations/?search=undefined&page=1"
                        onClick={[Function]}
                      >
                        First
                      </a>
                    </Link>
                  </li>
                </PaginationItem>
                <PaginationItem
                  disabled={true}
                  tag="li"
                >
                  <li
                    className="page-item disabled"
                  >
                    <Link
                      aria-label="Previous"
                      className="page-link"
                      replace={false}
                      to="#"
                    >
                      <a
                        aria-label="Previous"
                        className="page-link"
                        href="/"
                        onClick={[Function]}
                      >
                        «
                      </a>
                    </Link>
                  </li>
                </PaginationItem>
                <PaginationItem
                  disabled={false}
                  tag="li"
                >
                  <li
                    className="page-item"
                  >
                    <Link
                      aria-label="Next"
                      className="page-link"
                      replace={false}
                      to="/locations/?search=undefined&page=2"
                    >
                      <a
                        aria-label="Next"
                        className="page-link"
                        href="/locations/?search=undefined&page=2"
                        onClick={[Function]}
                      >
                        »
                      </a>
                    </Link>
                  </li>
                </PaginationItem>
                <PaginationItem
                  disabled={false}
                  tag="li"
                >
                  <li
                    className="page-item"
                  >
                    <Link
                      aria-label="Last"
                      className="page-link"
                      replace={false}
                      to="/locations/?search=undefined&page=2"
                    >
                      <a
                        aria-label="Last"
                        className="page-link"
                        href="/locations/?search=undefined&page=2"
                        onClick={[Function]}
                      >
                        Last
                      </a>
                    </Link>
                  </li>
                </PaginationItem>
              </ul>
            </nav>
          </Pagination>
        </div>
      </ListView>
    </div>
  </LocationsList>
</Router>
`;
